sql/core/src/main/scala/org/apache/spark/sql/execution/python/MapInBatchEvaluatorFactory.scala

```diff
@@ -36,7 +36,7 @@ class MapInBatchEvaluatorFactory(
     sessionLocalTimeZone: String,
     largeVarTypes: Boolean,
     pythonRunnerConf: Map[String, String],
-    pythonMetrics: Map[String, SQLMetric],
+    pythonMetrics: Option[Map[String, SQLMetric]],
```

Comment:
Why do we need to change this to Optional?

Reply:
In order to reuse `MapInBatchEvaluatorFactory` to read the data in executor side. We should integrate this to `Scan.reportDriverMetrics` though.

Reply:
Here: https://github.com/apache/spark/pull/44375